test/functional/test_runner.py

```diff
@@ -30,6 +30,9 @@
 
 os.environ["REQUIRE_WALLET_TYPE_SET"] = "1"
 
+# Should fail testing and immediately stop if free space falls below.
+MIN_FREE_SPACE = 1250 * 1024 * 1024
```

Comment:
![2024-02-23 14-08-26_Disk_usage](https://github.com/bitcoin/bitcoin/assets/581308/aaf81e7a-6ae5-4554-92ed-b3878a09c67b)

Thanks for the monitoring script you provided earlier @BrandonOdiwuor! Was able to get disk usage almost at 5GB using:
```
$ ./test/functional/test_runner.py --cachedir=/mnt/tmp/cache --tmpdir=/mnt/tmp --extended --jobs=16
```
So maybe some heuristic could be applied when running many jobs in parallel?
Default number of jobs is 4, so maybe something like
```python
MIN_FREE_SPACE = min(jobs, 4) * 320 * 1024 * 1024
```
That way we get `1280 * 1024 * 1024` for 4 jobs or less, and `5120 * 1024 * 1024` for 16 jobs.